### PR TITLE
feat: adding nginx configuration & gateway type in composite architecture 

### PR DESCRIPTION
**Pull request by neil85**
Adding nginx configuration & gateway type in composite architecture

### CODE REVIEW

File: charts/apisix-ingress-controller/templates/service-apisix.yaml [hunk 1]
```diff
@@ -17,4 +21,16 @@ spec:
...
   selector:
     {{- include "apisix-ingress-controller.selectorLabels" . | nindent 4 }}
-  type: NodePort
+  type: {{ .Values.gateway.type }}
+  externalTrafficPolicy: {{ .Values.gateway.externalTrafficPolicy }}
+  {{- if eq .Values.gateway.type "LoadBalancer" }}
+  {{- if .Values.gateway.loadBalancerIP }}
+  loadBalancerIP: {{ .Values.gateway.loadBalancerIP }}
+  {{- end }}
+  {{- if .Values.gateway.loadBalancerSourceRanges }}
```
Comment (lingsamuel) [inline charts/apisix-ingress-controller/templates/service-apisix.yaml:30]: it would be better to add some comments for these fields.

**Comment by lingsamuel** [comment]
`make helm-docs` is required

File: charts/apisix-ingress-controller/templates/service-apisix.yaml (hunk 1 above)
Comment (neil85) [reply charts/apisix-ingress-controller/templates/service-apisix.yaml:33]: updated comment.

**Comment by neil85** [comment]
> `make helm-docs` is required

done. thanks

File: charts/apisix-ingress-controller/templates/service-apisix.yaml (hunk 1 above)
Comment (lingsamuel) [reply charts/apisix-ingress-controller/templates/service-apisix.yaml:33]: Sorry for my unclear comment. I mean these fields should be in `values.yaml`, like `gateway.annotations`.

Comment (neil85) [reply charts/apisix-ingress-controller/templates/service-apisix.yaml:32]: ohh my bad. Updated values files as suggested

File: charts/apisix-ingress-controller/values.yaml [hunk 2]
```diff
@@ -209,3 +209,19 @@ securityContext: {}
   # readOnlyRootFilesystem: true
   # runAsNonRoot: true
   # runAsUser: 1000
+
```
Comment (lingsamuel) [inline charts/apisix-ingress-controller/values.yaml:212]: README file is auto-generated from field comments. Also a newline at eof is good.
```suggestion
gateway:
  # -- Apache APISIX service type for user access itself
  type: NodePort
  externalTrafficPolicy: Cluster
  # annotations:
  #   service.beta.kubernetes.io/aws-load-balancer-type: nlb
  # loadBalancerIP: a.b.c.d
  # loadBalancerSourceRanges:
  #   - "143.231.0.0/16"
  # -- load balancer ips
  externalIPs: []
  nginx:
    # -- Nginx workerRlimitNoFile
    workerRlimitNofile: "20480"
    # -- Nginx worker connections
    workerConnections: "10620"
    # -- Nginx worker processes
    workerProcesses: auto
    # -- Nginx error logs path
    errorLog: stderr
    # -- Nginx error logs level
    errorLogLevel: warn

```

File: charts/apisix-ingress-controller/values.yaml [hunk 3]
```diff
@@ -209,3 +209,27 @@ securityContext: {}
...
+    workerRlimitNofile: "20480"
+    # -- Nginx worker connections
+    workerConnections: "10620"
+    # -- Nginx worker processes
+    workerProcesses: auto
+    # -- Nginx error logs path
+    errorLog: stderr
+    # -- Nginx error logs level
+    errorLogLevel: warn
+
```
Comment (lingsamuel) [inline charts/apisix-ingress-controller/values.yaml:235]: Linter failed:
```log
charts/apisix-ingress-controller/values.yaml
  236:5     error    no new line character at the end of file  (new-line-at-end-of-file)
  236:1     error    trailing spaces  (trailing-spaces)
```

**Review by lingsamuel — APPROVED** [review]
Thanks!

**Review by AlinsRan — APPROVED** [review]

**Pull request merged**